Add an .editorconfig file for this repo

ScottGarman opened the pull request:
Specifies tab indent style for bash scripts and space indent style for python scripts.

ScottGarman commented:
I'm happy to add additional modeline options if desired.

mmlb commented:
I think it might be better to just setup an [editorconfig](https://editorconfig.org/) file instead this way it works for all editors.

mmlb commented:
@ScottGarman thoughts on my editorconfig comment?

mmlb reviewed (commented):
maybe better off as an editorconfig file instead?

ScottGarman committed:
Add an .editorconfig file for this repo

Specifies tab indent style for bash scripts and space indent style for
python scripts.

Signed-off-by: Scott Garman <sgarman@packet.com>

ScottGarman commented:
@mmlb I've finally gotten around to updating this PR to use editorconfig. Just specifying the indent style, which is the minimum needed to restore my sanity again.

mmlb commented:
@ScottGarman can I get you to remove the modelines in all the files present too? Don't want 2 sources for the settings.

ScottGarman committed:
Remove vim modelines

These will be replaced with an EditorConfig file.

Signed-off-by: Scott Garman <sgarman@packet.com>

ScottGarman commented:
@mmlb it turned out there was only one file we had (ci/Dockerfile) that had vim modelines set that could be trimmed out. There are several others that come in from the shunit submodule that aren't our code that I'm leaving out.
